Add sample maven/gradle projects.

itohro opened the pull request.

gs-rezaem commented on samples/reladomo-gradle-sample/gradle/wrapper/gradle-wrapper.properties:
```diff
@@ -0,0 +1,6 @@
+#Mon Sep 19 14:22:57 JST 2016
```
needs copyright/license

gs-rezaem commented on samples/reladomo-gradle-sample/gradlew:
```diff
@@ -0,0 +1,164 @@
+#!/usr/bin/env bash
+
```
needs copyright/license

gs-rezaem commented on samples/reladomo-gradle-sample/gradlew:
```diff
@@ -0,0 +1,164 @@
+#!/usr/bin/env bash
+
+##############################################################################
```
needs copyright/license

gs-rezaem commented on samples/reladomo-gradle-sample/gradlew.bat:
```diff
@@ -0,0 +1,90 @@
+@if "%DEBUG%" == "" @echo off
```
needs copyright/license

gs-rezaem commented on samples/reladomo-gradle-sample/src/main/resources/reladomo/db/OBJECT_SEQUENCE.ddl:
```diff
@@ -0,0 +1,8 @@
+drop table if exists OBJECT_SEQUENCE;
```
why are the db objects checked in?

gs-rezaem commented on samples/reladomo-gradle-sample/src/test/resources/test_data.txt:
```diff
@@ -0,0 +1,6 @@
+class sample.domain.Person
```
needs copyright/license

gs-rezaem commented on samples/reladomo-maven-sample/src/test/resources/test_data.txt:
```diff
@@ -0,0 +1,6 @@
+class sample.domain.Person
```
needs copyright/license

itohro commented:
You're right, no need to check in the generated ddl/idx files. I removed them and moved to generated-resources folder.

Added missing copyright/license info, too.

gs-rezaem commented:
The gradle copyright (Copyright 2007-2011 the original author or authors) seems nebulous and I'm not comfortable with that. I think we should not check in the wrapper files. We can add a note that says you must install gradle.

We won't need the notice file or the license files once we remove. Those are only necessary if we copy an artifact or link to (call) their code from our code.

Also, I don't see why we need two directories. The sample application might grow over time (e.g. add a sharding or bitemporal example). I think one directory with both build files is enough.

Finally, the db files are not resources. A resource is something that has to be on the classpath at runtime. I would put the generated files in target/generated-db and for a real project where the generated files are modified (permissions, table spaces, index hints, etc), I would check those in under src/main/db.

itohro commented:
Thanks for your comments!

- I got rid of the gradle wrapper. Added note about maven/gradle installation requirement in `README.md`, too.

- Regarding notice file, `HelloReladomoApp` calls Reladomo, sl4j and h2. Maybe we don't care Reladome itself but shouldn't we leave sl4j and h2 there? I left them for now. Please let me know.

- I consolidated them into a single sample directly and named `reladomo-sample-simple`. I agree we should add more samples. I'd think it's better to create separate directory for each functionality (e.g. `reladomo-sample-sharding`, `reladomo-sample-bitemporal` etc.) so that it's concise enough for new users to digest what's the minimum requirement for each functionality. We could then add `reladomo-sample-advanced` that has multiple functionalities combined together as more like a real life example.

- I made it to use `target/generated-db` for maven and `build/generated-db `for gradle. For this sample, these still need to be added as resource as the app loads ddl/idx in-memory. I added a commented in `pom.xml`/`build.gradle` to clarify such configuration is not meant to be done for typical production app.

gs-rezaem commented on samples/reladomo-sample-simple/NOTICE.txt:
```diff
@@ -0,0 +1,20 @@
+Reladomo Maven Sample Project
```
Not just maven anymore!

gs-rezaem commented on samples/reladomo-sample-simple/src/main/java/sample/HelloReladomoApp.java:
```diff
@@ -0,0 +1,92 @@
+    {
+        try
+        {
+            H2ConnectionManager.getInstance().prepareTables();
```
add a note that says this is not normal for prod applications

gs-rezaem commented on samples/reladomo-sample-simple/src/main/java/sample/HelloReladomoApp.java:
```diff
@@ -0,0 +1,92 @@
+        {
+            logger.info("Transaction Timeout is " + MAX_TRANSACTION_TIMEOUT);
+            MithraManager mithraManager = MithraManagerProvider.getMithraManager();
+            mithraManager.setTransactionTimeout(MAX_TRANSACTION_TIMEOUT);
```
add a note that notification should be configured here.

gs-rezaem commented on samples/reladomo-sample-simple/src/main/java/sample/domain/Person.java:
```diff
@@ -0,0 +1,46 @@
+		// You must not modify this constructor. Mithra calls this internally.
+		// You can call this constructor. You can also add new constructors.
+	}
+
```
add a domain specific method, like "getFullName()"

gs-rezaem commented on samples/reladomo-sample-simple/src/main/java/sample/util/H2ConnectionManager.java:
```diff
@@ -0,0 +1,124 @@
+        return schemaName;
+    }
+
+    // Load ddl/idx files under generated-db/sql in-memory.
```
add a note that not normal for prod

itohro commented:
Thanks for the quick feedback! Made those changes.
